go-sdk/datasources/prometheus/prometheus.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheus
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/perses/perses/go-sdk"
+	v1 "github.com/perses/perses/pkg/model/api/v1"
+	"github.com/perses/perses/pkg/model/api/v1/common"
+	"github.com/perses/perses/pkg/model/api/v1/datasource"
+	"github.com/perses/perses/pkg/model/api/v1/datasource/http"
+	"github.com/prometheus/common/model"
+	"github.com/sirupsen/logrus"
+)
+
+type PluginSpec struct {
+	DirectURL      *string         `json:"directUrl,omitempty" yaml:"directUrl,omitempty"`
+	Proxy          *http.Proxy     `json:"proxy,omitempty" yaml:"proxy,omitempty"`
+	ScrapeInterval *model.Duration `json:"scrapeInterval,omitempty" yaml:"scrapeInterval,omitempty"`
+}
+
+func NewDatasource(name string, url string) *DatasourceBuilder {
```

Comment:
I would avoid to set the `directURL` here since you have a method to set it later and it can be confusing about what URL is exactly set with this parameter